### PR TITLE
[builder] cleanup env dispatcher and Builder.build()

### PR DESCRIPTION
**Pull request by dnephin**
Cleanup some stale comments in dispatcher.env, and remove duplication around comparing env keys.

Factor out some functions from Builder.build()

Remove an unused field and method from Builder

### CODE REVIEW

File: builder/dockerfile/builder.go [hunk 1]
```diff
@@ -304,15 +294,27 @@ func (b *Builder) warnOnUnusedBuildArgs() {
...
 }
 
 // Cancel cancels an ongoing Dockerfile build.
-func (b *Builder) Cancel() {
-	b.cancel()
-}
+//func (b *Builder) Cancel() {
+//	clientCtx
+//	b.cancel()
+//}
```
Comment (aaronlehmann) [inline builder/dockerfile/builder.go]: Commented-out code?

Comment (dnephin) [inline builder/dockerfile/builder.go]: Oops, forgot to actually remove this. Thanks!

**Review by vdemeester — APPROVED** [review]
LGTM 🐯

**Review by tonistiigi — APPROVED** [review]
LGTM

**Pull request merged**